deploy/heroku-deploy/aws_cli_public_pgp_key.txt

```diff
@@ -0,0 +1,29 @@
+-----BEGIN PGP PUBLIC KEY BLOCK-----
```

Comment:
I don't think it should change, but I just checked and it does have an expiration date according to the [doc](https://docs.aws.amazon.com/cli/latest/userguide/getting-started-install.html#install-linux-verify):
```
Key ID:           A6310ACC4672
Type:             RSA
Size:             4096/4096
Created:          2019-09-18
Expires:          2023-09-17
User ID:          AWS CLI Team <aws-cli@amazon.com>
Key fingerprint:  FB5D B77F D5C1 18B8 0511  ADA8 A631 0ACC 4672 475C
```
I think the reason for an in-code text was to prevent man-in-the-middle attacks with the zip file, right? If we try using `curl` to get the public key, I think we'd still be vulnerable to those attacks. Not entirely sure though...

Reply:
I don't think a mitm attack here is feasible since it's over https, right? An attacker would need to have access to Amazon's certificate for the `awscli.amazonaws.com` domain in order to trick us into downloading a malicious file. I guess the idea here is that this is an extra check, so an attacker would need to get Amazon's cert _and_ the key they used to sign this thing. Is that the reasoning here? Are there other reasons to include this? (I've never been clear on how worried to be about accidental file integrity issues.)

Reply:
> I don't think a mitm attack here is feasible since it's over https, right?
I suppose so. A more likely (but still unlikely) possibility is `curl` using a modified `/etc/hosts` file, but I just found out that you need to pass in `--resolve` for curl to use virtual hosts. So... nvm!

In the AWS [doc](https://docs.aws.amazon.com/cli/latest/userguide/getting-started-install.html#install-linux-verify) for installing the CLI, there's an optional step to verify the `.zip` in case it is damaged/altered:
> The AWS CLI installer package .zip files are cryptographically signed using PGP signatures. If there is any damage or alteration of the files, this verification fails and you should not proceed with installation.

Even if the `.zip` is modified, we probably wouldn't be able to unzip it and install the CLI. This should throw an error in the `docker build` process, which the build script handles. In that case, I'll remove the PGP check so we don't have to keep a lookout for a new public key next year.